cmd/utils/flags.go

```diff
@@ -1165,27 +1100,27 @@ func SetP2PConfig(ctx *cli.Context, cfg *p2p.Config) {
 	setBootstrapNodesV5(ctx, cfg)
 
 	lightClient := ctx.GlobalString(SyncModeFlag.Name) == "light"
-	lightServer := (ctx.GlobalInt(LegacyLightServFlag.Name) != 0 || ctx.GlobalInt(LightServeFlag.Name) != 0)
+	lightServer := (ctx.GlobalInt(LightServeFlag.Name) != 0)
 
-	lightPeers := ctx.GlobalInt(LegacyLightPeersFlag.Name)
+	var lightPeers int
```

Comment:
you can do `lightPeers := ctx.GlobalInt(LightMaxPeersFlag.Name)` and drop the if altogether.

Reply:
Your PR is actually incorrect currently, because if the flag is not explicitly set, it won't use the default, rather will disable peering :)